pkg/sql/fingerprint_span.go

```diff
@@ -129,14 +128,26 @@ func (p *planner) fingerprintSpanFanout(
 		return 0, nil, err
 	}
 
-	var workerPartitions []SpanPartition
-	if len(spanPartitions) <= maxWorkerCount {
-		workerPartitions = spanPartitions
-	} else {
-		workerPartitions = make([]SpanPartition, maxWorkerCount)
-		for i, sp := range spanPartitions {
-			idx := i % maxWorkerCount
-			workerPartitions[idx].Spans = append(workerPartitions[idx].Spans, sp.Spans...)
+	maxLen := 0
+	count := 0
+	for _, partition := range spanPartitions {
+		length := len(partition.Spans)
+		maxLen = max(maxLen, length)
+		count += length
+	}
+	// Take as many workers as partitions to efficiently distribute work
+	maxWorkerCount = min(maxWorkerCount, count)
+
+	// Each span partition contains spans that are likely to be served by
+	// the same node. By round robin grabbing a span from each partition
+	// and pushing to the channel, ideally sequential workers won't attempt
+	// to read from the same node at the same time.
+	spanChannel := make(chan roachpb.Span, count)
```

Comment:
I wonder if we'd benefit from shuffling the partitions too?

Reply:
With the round robin setup, what would shuffling improve?